### DISCUSSION TITLE
Flow operation Condition is not working

### DISCUSSION THREAD

**Discussion by zhex900** [directus/directus#14108] [Archive]
### Describe the Bug

I have a `Condition` that resolves to `false`, but it continues in the operation. It should break. What am I missing?

### To Reproduce

Look at my state diagram. 
![Screen Shot 2022-06-25 at 7 32 16 am](https://user-images.githubusercontent.com/5380757/175696189-0a69fd70-6794-42eb-9b71-ce693e0ee82a.png)

![Screen Shot 2022-06-25 at 7 32 37 am](https://user-images.githubusercontent.com/5380757/175696212-c94e0de3-6e0a-4e87-8a57-66e3bc0084d6.png)

**Comment by azrikahar** [comment]
I believe you should remove the wrapping `filter`, as your incoming payload isn't contained inside `filter` property. (not to be confused with filter used in queries)

Also refer to this comment on why not to wrap `$last` in brackets in this case: https://github.com/directus/directus/discussions/13851#discussioncomment-2939405

Perhaps you can try this condition instead:

```json
{
   "$last": {
       "state": {
          "_neq": "error"
       }
   }
}
```

**Reply by zhex900** [reply]
I didn't add the `filter` key word.

This is the condition 

```
{
    "$trigger": {
        "key": {
            "_eq": 222
        }
    }
}
```
<img width="692" alt="Screen Shot 2022-06-26 at 6 42 48 am" src="https://user-images.githubusercontent.com/5380757/175789943-1c442759-2c7f-4084-b8b5-62767707155b.png">

I got this error.

```
[
  {
    "_original": {
      "$trigger": {
        "payload": {
          "reason": [
            "update"
          ]
        },
        "key": 51,
        "collection": "publish"
      },
      "$last": {
        "key": "51"
      },
      "$accountability": {
        "user": "a7f4a6cf-5f19-4a41-a3f7-ca9fa18dbc16",
        "role": "8488f195-a3a3-4540-8739-d638ee100045",
        "admin": true,
        "app": true,
        "ip": "120.159.43.173",
        "userAgent": "Amazon CloudFront",
        "permissions": []
      },
      "t4333": {
        "key": "51"
      }
    },
    "details": [
      {
        "message": "\"$last.key\" is not allowed",
        "path": [
          "$last",
          "key"
        ],
        "type": "object.unknown",
        "context": {
          "child": "key",
          "label": "$last.key",
          "value": "51",
          "key": "key"
        }
      }
    ]
  }
]
```

**Comment by licitdev** [comment]
This looks to be a duplicate of issue #13855.
It should be fixed via PR #13596.
